lib/host/wasi/inode-linux.cpp

```diff
@@ -966,6 +966,84 @@ WasiExpect<void> INode::sockShutdown(__wasi_sdflags_t SdFlags) const noexcept {
   return {};
 }
 
+WasiExpect<void> INode::sockGetError() const noexcept {
+  int status;
+  socklen_t slen;
+
+  if (auto Res = ::getsockopt(Fd, SOL_SOCKET, SO_ERROR, (void *) &status, &slen); unlikely(Res < 0)) {
```

Comment:
`getsockopt` can have several different opt queries. We would make every query to be a standalone function. Please modify your function signature to fit the `getsockopt`.

Reply:
Also, I am not sure if we really need this. If the errno is provided in each wasi socket function.

Reply:
https://github.com/WebAssembly/wasi-libc/blob/659ff414560721b1660a19685110e484a081c3d4/libc-bottom-half/cloudlibc/src/libc/poll/poll.c#L92-L100

when I poll some socket, poll() drop errno. maybe I can rewrite a poll.

Reply:
We can also keep this function. However, it should follow the signature of `getsockopt`. In case that we will need to query the other options in the future.